### PR TITLE
Favorite mods in enable/disable mods list

### PR DESCRIPTION
**Pull request by mayromr**
Started working on the changes suggested at:  #501.
TODO:

- [x] Add keyboard hints (or whatever the little key indicator are called).
- [x] Add localizations.
- [x] Test with nested mod dependencies.
- [X] Handle edge case where cyclic mod dependencies will loops forever ( not sure if its even possible to load with Everest ).

Feedback would be appreciated.

### CODE REVIEW

**Comment by mayromr** [comment]
Basic functionalities are done, but this is a huge mess because I am new to celeste modding.
I would love if someone could review this.

File: Celeste.Mod.mm/Mod/UI/TextMenuExt.cs [hunk 1]
```diff
@@ -1353,5 +1353,93 @@ public class BatchModeContext : IDisposable {
             }
         }
 
+        public class SubMenuWithInputs : TextMenu.Item {
```
Comment (mayromr) [inline Celeste.Mod.mm/Mod/UI/TextMenuExt.cs]: Might be worth to try and implement a `EaseInDecorator` in order to reduce code repetition.
something like this:

``` c#
public class EaseInDecorator<T> : TextMenu.Item, IItemExt where T : TextMenu.Item, IItemExt {
            private T inner;

            public EaseInDecorator(T inner) {
                this.inner = inner;
            }

}
```
